lib/Tomcat/Utils.pm

```diff
@@ -0,0 +1,97 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Provide functionality for tomcat installation and configuration,
+# accessing the tomcat manager page, and browsing tomcat examples using keyboard.
+# Maintainer: George Gkioulis <ggkioulis@suse.com>
+
+
+package Tomcat::Utils;
+use base "x11test";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+use registration;
+
+# allow a 60 second timeout for asserting needles
+use constant TIMEOUT => 60;
+
+# Use keyboard to browse the examples faster
+sub browse_with_keyboard {
+    my ($self, $fallback_needle, $test_func, $tab_num) = @_;
+
+    assert_screen($fallback_needle, TIMEOUT);
+    for (1 .. $tab_num) { send_key('tab'); }
+    send_key('ctrl-ret');
+    send_key('ctrl-tab');
+
+    $test_func->();
+
+    send_key('ctrl-w');
+}
+
+# Install tomcat and set initial configuration
+sub tomcat_setup() {
+    my $tomcat_users_xml = <<'EOF';
+<?xml version="1.0" encoding="UTF-8"?>
+<tomcat-users xmlns="http://tomcat.apache.org/xml"
+              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:schemaLocation="http://tomcat.apache.org/xml tomcat-users.xsd"
+              version="1.0">
+<role rolename="manager-gui"/>
+<user name="admin" password="admin" roles="manager-gui,tomcat"/>
+</tomcat-users>
+EOF
+
+    # log in to root console
+    select_console('root-console');
+
+    record_info('Initial Setup');
+
+    # enable Web and Scripting Module for SLE 15 and above
+    if (is_sle('>=15')) {
+        add_suseconnect_product("sle-module-web-scripting");
```

Comment:
I am still not sure about this line. Isn't the preferred method using `SCC_ADDONS=serverapp,script`?

Reply:
I 'm on it!

Reply:
I'm sure QAM tests have `SCC_ADDONS=serverapp,script` but tests on developed products not.

Reply:
So maybe using `unless is_released` could be the right way to go?